docs/multicall-notes.md

```diff
@@ -0,0 +1,226 @@
+
+Base fee per gas is not adjusted in the phantom blocks, their base fee remains constant.
+
+## Default values for transactions
+As multicall is an extension to `eth_call` we want to enable the nice user experience that the user does not need to provide all required values for a transaction. We are assuming following defaults if the variable is not provided by the user:
+| parameter name | description |
+-----------------|-----------------------
+| type | `0x2` |
+| nonce | Defaults to correct nonce |
+| to | `null` |
+| from | `0x0000000000000000000000000000000000000000` |
+| gas limit | Remaining gas in the current block. This is calculated dynamically one by one for each transaction that is being processed. |
+| value | `0x0` |
+| input | no data |
+| gasPrice | `0x0` |
+| maxPriorityFeePerGas | `0x0` |
+| maxFeePerGas | `0x0` |
+| accessList | empty array |
+
+## Overriding default values
+The default values of blocks and transactions can be overriden. For Transactions we allow overriding of variables `type`, `nonce`, `to`, `from`, `gas limit`, `value`, `input`, `gasPrice`, `maxPriorityFeePerGas`, `maxFeePerGas`, `accessList`, and for blocks we allow modifications of `number`, `time`, `gasLimit`, `feeRecipient`, `prevRandao` and `baseFeePerGas`:
+```json
+"blockOverrides": {
+	"number": "0x14",
+	"time": "0xc8",
+	"gasLimit": "0x2e631",
+	"feeRecipient": "0xc100000000000000000000000000000000000000",
+	"prevRandao": "0x0000000000000000000000000000000000000000000000000000000000001234",
+	"baseFeePerGas": "0x14"
+},
+```
+All the other fields are computed automatically (eg, `stateRoot` and `gasUsed`) or kept as their default values (eg. `uncles` or `withdrawals`). When overriding `number` and `time` variables for blocks, we automatically check that the block numbers and time fields are strictly increasing (we don't allow decreasing, or duplicated block numbers or times). If the block number is increased more than `1` compared to the previous block, phantom blocks are created to fill the gaps.
+
+An interesting note here is that an user can specify block numbers and times of some blocks, but not for others. When block numbers of times are left unspecified, the default values will be used. After the blocks have been constructed, and default values are calculated, the blocks are checked that their block numbers and times are still valid.
+
+## ETH transfer logs
+When `traceTransfers` setting is enabled on `eth_multicallV1` The multical will return logs for ethereum transfers along with the normal logs sent by contracts. The ETH transfers are identical to ERC20 transfers, except the "sending contract" is address `0x0`.
```

Comment:
I'd suggest using `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE` as the sending contract, since that's become a de facto approach for representing ETH as a token address within contracts.

Reply:
I agree a search for the zero address would yield more results, but only because the zero address is used for a much wider range of cases, whereas `0xEee...` seems to only be used to represent ETH. This is anecdotal and it's hard to know concretely, but in my experience `0xEee...` is more common for representing ETH than the zero address. 

Personally, if I didn't know about the special ETH logs, seeing `0xEee...` would make the log's meaning more intuitive and apparent than seeing `0x000...`.

Reply:
> I agree a search for the zero address would yield more results, but only because the zero address is used for a much wider range of cases, whereas `0xEee...` seems to only be used to represent ETH. This is anecdotal and it's hard to know concretely, but in my experience `0xEee...` is more common for representing ETH than the zero address.
> 
I kinda agree with @mds1 that `0xEee..` seems more appropriate for Eth transfer logs

Reply:
I believe the spec has since been updated to use `0xEee....`.

Reply:
The notes still mentions `0x0` here: https://github.com/ethereum/execution-apis/pull/484/files#diff-1e4727bbfb75c5cc85054abbf972a1cd472e9e512a9d8bbc8152210a0062182bR116

Aside: It still feels "unclean" to me using a specific non-reserved address, but in researching this I was surprised to learn that EIP-1352 was never officially adopted.

Reply:
`0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE` is not correct if you're using EIP-1191, the correct chainId encoded address would be `0xeeeEEEeEEeeeEeEeEEEeEeeeeEEEEeEEeEeeeeeE`

having it as `0x00...` would be agnostic to chainId encoding.

Reply:
I and others have argued against EIP-1191 in the discussion for it: https://github.com/ethereum/EIPs/issues/1121

My position is still largely the same.  It is backward incompatible with EIP-55 checksumming, so I generally don't recommend apps checksum to 1191 until long after almost all tooling supports *reading* 1191 checksums.